test: Run tests against django main branch

marksweb opened the pull request:
This introduces testing, allowed to fail, against the main branch of django so we have visibility of what is coming in future versions.

fsbraun commented on .github/workflows/test.yml:
```diff
@@ -155,3 +155,135 @@ jobs:
+
+    services:
+      postgres:
+        image: postgres:12
```
```suggestion
        image: postgres:13
```
Django main dropped postgres 12

fsbraun commented on .github/workflows/test.yml:
```diff
@@ -155,3 +155,135 @@ jobs:
+
+    - name: Test with django test runner
+      run: |
+        python manage.py test
```
```suggestion
        python -c "from django import __version__ ; print(f'Django version {__version__}')"
        python manage.py test
``` 
For transparency let's print out which version the django main had...

fsbraun commented on .github/workflows/test.yml:
```diff
@@ -155,3 +155,135 @@ jobs:
+
+    - name: Test with django test runner
+      run: |
+        python manage.py test
```
```suggestion
        python -c "from django import __version__ ; print(f'Django version {__version__}')"
        python manage.py test
```
Same here.

fsbraun reviewed (changes requested):
Lines 184 and 185 need to be swapped. Otherwise, the requirements will uninstall the django main branch if there is a `<5` somewhere.

Like wise 232 and 233, and 280 and 281

fsbraun commented on .github/workflows/test.yml:
```diff
@@ -155,3 +155,137 @@ jobs:
+    - name: Test with django test runner
+      run: |
+        python -c "from django import __version__ ; print(f'Django version {__version__}')"
+        python -c "from django import __version__ ; print(f'Django version {__version__}')"
```
```suggestion
```

marksweb commented:
@fsbraun have we removed the Django 5 restriction from develop 4? Could do with updating my branches before merging to use it if we've dropped that limit.

fsbraun commented:
Just did. But also need to fix ruff... One minute.
